File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cs
using System;

public class Solution {
    public string solution(string video_len, string pos, string op_start, string op_end, string[] commands) {
        int len  = ToSeconds(video_len);
        int cur  = ToSeconds(pos);
        int opS  = ToSeconds(op_start);
        int opE  = ToSeconds(op_end);
        
        if(cur >= opS && cur <= opE) cur = opE;
        foreach(var cmd in commands)
        {
            if(cmd == "next")
            {
                cur = Math.Min(len, cur + 10);
            }
            else if(cmd == "prev")
            {
                cur = Math.Max(0, cur - 10);
            }
            
            if(cur >= opS && cur <= opE) cur = opE;
        }
        return ToMMSS(cur);
    }
    
    private int ToSeconds(string mmss)
    {
        var parts = mmss.Split(':');
        int m = int.Parse(parts[0]);
        int s = int.Parse(parts[1]);
        return m * 60 + s;
    }
    private string ToMMSS(int time)
    {
        int m = time / 60;
        int s = time % 60;
        return $"{m:D2}:{s:D2}";
    }
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cpp
#include <string>
#include <vector>

using namespace std;

int toSec(string time)
{
    int m = stoi(time.substr(0, 2));
    int s = stoi(time.substr(3, 2));
    
    return m * 60 + s;
}
string toString(int time)
{
    int m = time / 60;
    int s = time % 60;
    
    string mm = (m < 10 ? "0" : "") + to_string(m);
    string ss = (s < 10 ? "0" : "") + to_string(s);
    
    return mm + ":" + ss;
}
string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {
    
    int video = toSec(video_len);
    int cur = toSec(pos);
    
    int opS = toSec(op_start);
    int opE = toSec(op_end);
    
    for(string cmd : commands)
    {
        if(opS <= cur && cur <= opE)
            cur = opE;
        
        if(cmd == "next")
        {
            cur += 10;
            if(cur > video)
                cur = video;
        }
        else
        {
            cur -= 10;
            if(cur < 0)
                cur = 0;
        }
        
        if(opS <= cur && opE >= cur)
            cur = opE;
    }
    return toString(cur);
}